src/config.cpp

```diff
@@ -167,7 +174,7 @@ static void reset() {
 	core_opts = core();
 
 	pattern_opts.clear();
-	source.reset(new rapidjson::Document());
+	get_source().reset(new rapidjson::Document());
```

Comment:
This will actually change only the copy of the `source` instance that was returned from the `get_source()` function.